projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-endpoint.php

```diff
@@ -1,70 +1,108 @@
+require JETPACK__PLUGIN_DIR . '/modules/module-info.php';
 
 /**
  * Base class for Jetpack Endpoints, has the validate_call helper function.
  */
 abstract class Jetpack_JSON_API_Endpoint extends WPCOM_JSON_API_Endpoint {
 
+	/**
+	 * Needed capabilities.
+	 *
+	 * @var string
+	 */
 	protected $needed_capabilities;
+
+	/**
+	 * Expected actions.
+	 *
+	 * @var array
+	 */
 	protected $expected_actions = array();
-	protected $action;
 
+	/**
+	 * The action.
+	 *
+	 * @var string
+	 */
+	protected $action;
 
+	/**
+	 * Callback function.
+	 *
+	 * @param string $path - the path.
+	 * @param int    $blog_id - the blog ID.
+	 * @param object $object - parameter is for making the method signature compatible with its parent class method.
+	 */
 	public function callback( $path = '', $blog_id = 0, $object = null ) {
-		if ( is_wp_error( $error = $this->validate_call( $blog_id, $this->needed_capabilities ) ) ) {
+		$error = $this->validate_call( $blog_id, $this->needed_capabilities );
+		if ( is_wp_error( $error ) ) {
 			return $error;
 		}
 
-		if ( is_wp_error( $error = $this->validate_input( $object ) ) ) {
+		$error = $this->validate_input( $object );
+		if ( is_wp_error( $error ) ) {
 			return $error;
 		}
 
 		if ( ! empty( $this->action ) ) {
-			if( is_wp_error( $error = call_user_func( array( $this, $this->action ) ) ) ) {
+			$error = call_user_func( array( $this, $this->action ) );
+			if ( is_wp_error( $error ) ) {
 				return $error;
 			}
 		}
 
 		return $this->result();
 	}
 
+	/**
+	 * The result function.
+	 */
 	abstract protected function result();
 
-	protected function validate_input( $object ) {
+	/**
+	 * Validate input.
+	 *
+	 * @param object $object - unused, for parent class compatability.
+	 */
+	protected function validate_input( $object ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
 		$args = $this->input();
 
-		if( isset( $args['action'] ) && $args['action'] == 'update' ) {
+		if ( isset( $args['action'] ) && $args['action'] === 'update' ) {
 			$this->action = 'update';
 		}
 
-		if ( preg_match( "/\/update\/?$/", $this->path ) ) {
+		if ( preg_match( '/\/update\/?$/', $this->path ) ) {
 			$this->action = 'update';
 
-		} elseif( preg_match( "/\/install\/?$/", $this->path ) ) {
+		} elseif ( preg_match( '/\/install\/?$/', $this->path ) ) {
 			$this->action = 'install';
 
-		} elseif( ! empty( $args['action'] ) ) {
-			if( ! in_array( $args['action'], $this->expected_actions ) ) {
+		} elseif ( ! empty( $args['action'] ) ) {
+			if ( ! in_array( $args['action'], $this->expected_actions, true ) ) {
 				return new WP_Error( 'invalid_action', __( 'You must specify a valid action', 'jetpack' ) );
 			}
-			$this->action =  $args['action'];
+			$this->action = $args['action'];
 		}
 		return true;
 	}
 
 	/**
 	 * Switches to the blog and checks current user capabilities.
-	 * @return bool|WP_Error a WP_Error object or true if things are good.
```

Comment:
Because for some reason, my brain could have sword phpcs complained about return statements once upon a time (which is why I haven't added them as well) but clearly I got some kinda wires crossed here.